app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    results = {}
+
+    def wrapper(*args) -> str:
```

Comment:
You could apply a decorator from `functools` to preserve the wrapped function name and docstring.